src/builtins/BuiltinTemporal.cpp

```diff
@@ -206,6 +390,85 @@ static Value builtinTemporalCalendarPrototypeDay(ExecutionState& state, Value th
     return Value(TemporalCalendar::ISODay(state, temporalDateLike));
 }
 
+static Value builtinTemporalCalendarPrototypeDayOfWeek(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    TemporalPlainDate* temporalDate = TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
+    Value epochDays = DateObject::makeDay(state, Value(temporalDate->year()), Value(temporalDate->month() - 1), Value(temporalDate->day()));
+    ASSERT(std::isfinite(epochDays.asNumber()));
+    int dayOfWeek = DateObject::weekDay(DateObject::makeDate(state, epochDays, Value(0)).toUint32(state));
+    return Value(dayOfWeek == 0 ? 7 : dayOfWeek);
+}
+
+static Value builtinTemporalCalendarPrototypeDayOfYear(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    TemporalPlainDate* temporalDate = TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
+    Value epochDays = DateObject::makeDay(state, Value(temporalDate->year()), Value(temporalDate->month() - 1), Value(temporalDate->day()));
+    ASSERT(std::isfinite(epochDays.asNumber()));
+
+    return Value(TemporalCalendar::dayOfYear(state, epochDays));
+}
+
+static Value builtinTemporalCalendarPrototypeWeekOfYear(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalCalendar(state, thisValue, argc);
+
+    TemporalPlainDate* temporalDate = TemporalPlainDate::toTemporalDate(state, argv[0]).asObject()->asTemporalPlainDateObject();
```

Comment:
Would it be a good idea to add a variant of checkTemporalCalendar, which also returns with a temporalDate?